java/Ice/middleware/server/src/main/java/com/example/ice/middleware/server/AuthorizationMiddleware.java

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) ZeroC, Inc.
+
+package com.example.ice.middleware.server;
+
+import com.zeroc.Ice.DispatchException;
+import com.zeroc.Ice.IncomingRequest;
+import com.zeroc.Ice.OutgoingResponse;
+import com.zeroc.Ice.ReplyStatus;
+import com.zeroc.Ice.UserException;
+import java.util.concurrent.CompletionStage;
+
+/**
+ * The AuthorizationMiddleware intercepts incoming requests and checks the token entry in their contexts.
+ */
+class AuthorizationMiddleware implements com.zeroc.Ice.Object {
+    private final com.zeroc.Ice.Object _next;
+    private final String _validToken;
+
+    public CompletionStage<OutgoingResponse> dispatch(IncomingRequest request) throws DispatchException, UserException {
+        String token = request.current.ctx.get("token");
+
+        // Check if the request has a valid token in its context.
+        if (token == null || !token.equals(_validToken)) {
+            System.out.println("Rejecting request with invalid token " + token);
+            throw new DispatchException(ReplyStatus.Unauthorized.value(), "Invalid token " + token);
+        }
+
+        System.out.println("Accepting request with token " + token);
+        
+        // Continue the dispatch pipeline.
+        return _next.dispatch(request);
+
+    }
```

Comment:
Seems like an extra newline here:
```suggestion
        return _next.dispatch(request);
    }
```